azkaban-common/src/main/java/azkaban/jobExecutor/ProcessJob.java

```diff
@@ -203,16 +205,21 @@ public void run() throws Exception {
     String effectiveUser = null;
     final boolean isExecuteAsUser = this.sysProps.getBoolean(EXECUTE_AS_USER, true);
 
+    //Get list of users we never execute flows as. (ie: root, azkaban)
+    final String [] userBlackList = this.sysProps.getString(BLACK_LISTED_USERS, "root,azkaban").split(",");
+
     // nativeLibFolder specifies the path for execute-as-user file,
     // which will change user from Azkaban to effectiveUser
     if (isExecuteAsUser) {
       final String nativeLibFolder = this.sysProps.getString(NATIVE_LIB_FOLDER);
       executeAsUserBinaryPath =
           String.format("%s/%s", nativeLibFolder, "execute-as-user");
       effectiveUser = getEffectiveUser(this.jobProps);
-      if ("root".equals(effectiveUser)) {
-        throw new RuntimeException(
-            "Not permitted to proxy as root through Azkaban");
+      for (String blackListedUser : userBlackList) {
```

Comment:
Just make `userBlackList` a `Set` here and do a `contains()` check. 
nit: probably `blackListedUsers` instead of `userBlackList`?